### ISSUE TITLE
unhandled file extension 'las'

### ISSUE THREAD

**Issue by adamsuh008** [CloudCompare/CloudCompare#1050]
[Describe the bug]
Hi, I'm trying to build Cloudcompare and make my own plugins.
However, after I build Cloudcompare with PDAL plugin, las file format is not handled.
I thought I followed the correct step to support las format, but it's not working.
Since I'm a beginner in programming, I need some help...

[How to reproduce]
1. git clone --recursive cloudcompare
2. ccmake
COMPILE_CC_CORE_LIB_WITH_TBB -required with parallel processing
PLUGIN_IO_QCORE
PLUGIN_IO_QPDAL
3. configure, generate
4. make all
5. make install
6. Cloudcompare (in terminal)
7. double click executable in build/qCC (looks different from 6, don't know why...)

[Expected behaviour]
Able to handle las file format

[My environment]
 - OS & Version: Ubuntu 16.04
 - Graphics card: NVIDIA GeForce RTX 2080
 - Qt Version: 5.5.1

**Comment by tmontaigu** [comment]
In CloudCompare log console, is there any message about failing to load the PLUGIN_IO_QPDAL ?

I see that you use Ubuntu 16.04 which does not have pdal in the official repositories, how did you install pdal ?

It's probably that some dependencies (.so) of pdal are not found when CloudCompare loads the plugin.

**Comment by dgirardeau** [comment]
Not sure how it works on Linux, but normally you should run the version which is in the CMake 'install' directory (CMAKE_INSTALL_PREFIX) and not the one in the build folder...

**Comment by adamsuh008** [comment]
@tmontaigu I installed pdal through conda in my virtual environment. For adding the pdal directory, I just wrote /anaconda3/envs/lib/cmake/PDAL. Also, there wasn't any message about failing to load the PLUGIN_IO_PDAL. The program was built fine, but did not support PDAL.

**Comment by adamsuh008** [comment]
@dgirardeau 6.Cloudcompare(in terminal) is supposed to run the version which is in the CMake 'install' directory, but I was just wondering why it was different from the one in the build folder...

**Comment by tmontaigu** [comment]
make sure that when you run cloudcompare from the terminal your anaconda environment is activated

**Comment by adamsuh008** [comment]
@tmontaigu That seemed like what I was missing. However, when I erased the file and re-installed & built Cloudcompare, I faced another error message...

QSocketNotifier: Can only be used with threads started with QThread
Segmentation fault (core dumped)

I think I did nothing different from the past. Don't know why that happened...

**Comment by adamsuh008** [comment]
@tmontaigu I tried to re-build Cloudcompare and it looks different again.

QSocketNotifier: Can only be used with threads started with QThread << still pops out.

However, now I can run Cloudcompare with 'libQEDL_PLUGIN.so does not supply meta data in the Q_PLUGIN_METADATA(see one of the example plugins).' error message.
Also, las file format is still not supported with anaconda environment activated.

**Comment by adamsuh008** [comment]
I wonder if anyone with Ubuntu 16.04 can run Cloudcompare with las format supported...

**Comment by adamsuh008** [comment]
![Screenshot from 2020-03-02 14-52-52](https://user-images.githubusercontent.com/54871715/75649137-8d604300-5c95-11ea-982c-0f6a2560a65a.png)

this one seems to be the problem

**Comment by dgirardeau** [comment]
Is PDAL linked as a .so file as well? Is it accessible to anyone? (not sure how it works on Linux, but it looks similar to the error we get on Windows when a missing DLL dependency prevents Qt from loading the plugin)

**Comment by adamsuh008** [comment]
![image](https://user-images.githubusercontent.com/54871715/75733571-08c70080-5d39-11ea-95d5-01c925686619.png)

this is my cmake file setting

![image](https://user-images.githubusercontent.com/54871715/75733623-2c8a4680-5d39-11ea-9209-41ad50174131.png)

this is what I get for make install. PDAL plugin does not show up.

![image](https://user-images.githubusercontent.com/54871715/75733672-462b8e00-5d39-11ea-94e9-b477b731a8a1.png)
![image](https://user-images.githubusercontent.com/54871715/75733692-504d8c80-5d39-11ea-9f35-07e5599a05e2.png)

this is when I run Cloudcompare... every time I delete everything and try again, I'm getting a new error...

**Comment by adamsuh008** [comment]
After three changes, now it works perfectly!

1. build PDAL from sourcecode (github)
2. turnoff anaconda completely
3. delete everything related to cloudcompare and start building again

![image](https://user-images.githubusercontent.com/54871715/75735003-d0c1bc80-5d3c-11ea-9077-ad7674404a20.png)
Although these messages appear, all the functions are working.
Thank you very much for the help!

**Comment by dgirardeau** [comment]
Thanks for the feedback!

**Comment by alexwangxiang** [comment]
> ![Screenshot from 2020-03-02 14-52-52](https://user-images.githubusercontent.com/54871715/75649137-8d604300-5c95-11ea-982c-0f6a2560a65a.png)
> 
> this one seems to be the problem

I encountered with the same problem, and I also built pdal from sourcecode. For me, the keypoint is that libpdal_base.so.13 is installed in /usr/local/lib, which is not included in $LD_LIBRARY_PATH. I added /usr/local/lib to the bottom of /etc/ld.so.conf, and run the command 'sudo ldconfig'. Then I opened CloudCompare through terminal, and qPDAL_IO_PLUGIN can be loaded successfully this time.
